csrc/kernel.cpp

```diff
@@ -32,6 +32,21 @@ class KernelIrScanner : private IrVisitor {
  public:
   explicit KernelIrScanner(const Kernel* kernel) {
     index_type_ = kernel->indexType();
+
+    summary_.stream_parallelized = [&]() {
+      for (const auto& ios : {kernel->inputs(), kernel->outputs()}) {
```

Comment:
A brainstorm kinda question just for fun.

We are limiting the check only on IO tensors. Conceptually, we are not supporting intermediate buffers with stream id? The made-up scenario I'm thinking about is using stream id on rfactor for reduction.

Reply:
That's right. Are you trying to confirm your understanding or are you trying to point to a problem in the code?

Reply:
Just trying to clarify the protocol for my own education. 🫡